docs: add forge-overview.md (#35473)
electron/electron: docs/tutorial/forge-overview.md

# Distributing Apps With Electron Forge

> Electron Forge is a complete tool for creating, publishing, and installing modern Electron applications. Electron Forge unifies the existing (and well maintained) build tools for Electron development into a simple, easy to use package so
> that anyone can jump right in to Electron development.

Reviewer: ```suggestion
Electron Forge is a complete tool for creating, publishing, and installing modern Electron applications. Electron Forge unifies Electron's build tooling ecosystem into a single extensible interface so
that anyone can jump right in to Electron development.
```

Some copy edits:
* Didn't want to confuse `build tools` with `@electron/build-tools`
* `(and well maintained)` is a bit unnecessary 😂 
* Forge is not a single package, it's a monorepo with dozens of them!

> Electron Forge is a complete tool for creating, publishing, and installing modern Electron applications. Electron Forge unifies the existing (and well maintained) build tools for Electron development into a simple, easy to use package so
> that anyone can jump right in to Electron development.

Reviewer 2: :+1: on @erickzhao's suggestions, piggybacking on them here

```suggestion
Electron Forge is a tool for packaging and publishing Electron applications.
It unifies Electron's tooling ecosystem into a single extensible interface so
that anyone can jump right in to making Electron apps.
```

Minor changes:

- "modern" in "modern Electron applications" is implied
- "creating" is a little ambiguous; it _could_ be read as Forge also being an IDE. Changed to "packaging"
- 'Electron development" -> clarify that we're talking about app development, not Electron maintenance
- "Complete tool" - might just be me, but that phrasing didn't scan right to me

Author: I agree with both of you! Similar wording is used elsewhere in our documentation: I'll make changes there as well to modernize

## Getting started

> The [Electron Forge Docs] contain detailed information that can assist you through the process
> of taking your application from source code to your end users' machines.
> This process contains several subtasks, including:

Reviewer: ```suggestion
The [Electron Forge docs] contain detailed information on
taking your application from source code to your end users' machines.
This includes:
```

Making this shorter

* Packaging your application [(package)]

> * Generating distributable files for each OS [(make)], and,

Reviewer: ```suggestion
* Generating executables and installers for each OS [(make)], and,
```

I think `distributable` isn't as common of a term

* Publishing these files to online platforms to download [(publish)].

> For beginners, we recommend following through the cumulative Electron [tutorial] to develop, build, package and publish your first Electron app. If you have already developed an app on your machine and want to start on Forge-specific steps, start from [step 5] of the tutorial.

Reviewer: ```suggestion
For beginners, we recommend following through Electron's [tutorial] series to develop, build, package and publish your first Electron app. If you have already developed an app on your machine and want to start on packaging and distribution, start from [step 5] of the tutorial.
```

## Getting help

* If you need help with developing your app, our [community Discord server][discord] is a great place to get advice from other Electron app developers.
* If you suspect you're running into a bug with Forge, please check the [GitHub issue tracker] to see if any existing issues match your problem. If not, feel free to fill out our bug report template and submit a new issue.

[Electron Forge Docs]: https://www.electronforge.io/
[step 5]: ./tutorial-5-packaging.md
[(package)]: https://www.electronforge.io/cli#package
[(make)]: https://www.electronforge.io/cli#make
[(publish)]: https://www.electronforge.io/cli#publish
[GitHub issue tracker]: https://github.com/electron-userland/electron-forge/issues

> [discord]: https://discord.gg/electronjs

Reviewer 2: I don't mind either way, but am curious why the non-vanity would be preferred? `/electronjs` seems inherently more readable

> [discord]: https://discord.gg/electronjs

Reviewer: It is, but using a non-vanity is helpful in two ways:
* We've changed the vanity URL in the past and it's a pain to propagate
* We can track the % of engagement from our docs/website better by using a separate link

# Distributing Apps With Electron Forge

## Introduction to Electron Forge

Electron Forge is a complete tool for creating, publishing, and installing modern Electron applications. Electron Forge unifies the existing (and well maintained) build tools for Electron development into a simple, easy to use package so
that anyone can jump right in to Electron development.
This allows developers to take their app from source code to their end users' machines.

## Getting started

The [Electron Forge Docs] contain detailed information that can assist you through the process
of taking your application from source code to your end users' machines.
This process contains several subtasks, including:

* Packaging your application [(package)]
* Generating distributable files for each OS [(make)], and,
* Publishing these files to online platforms to download [(publish)].

> We recommend following through the cumulative Electron [tutorials] to develop, build, package and publish your first Electron app. If you have already developed an app on your machine and want to start on forge-specific steps, start from [step 5] of the tutorial.

Reviewer: ```suggestion
For beginners, we recommend following through the Electron [tutorial] to develop, build, package and publish your first Electron app. If you have already developed an app on your machine and want to start on Forge-specific steps, start from [step 5] of the tutorial.
```

## Getting help

* If you need help with developing your app, our [community Discord server][discord] is a great place to get advice from other Electron app developers.

> * If you suspect you're running into a bug with the electron-forge package, please check the [GitHub issue tracker] to see if any existing issues match your problem. If not, feel free to fill out our bug report template and submit a new issue.

Reviewer: ```suggestion
* If you suspect you're running into a bug with Forge, please check the [GitHub issue tracker] to see if any existing issues match your problem. If not, feel free to fill out our bug report template and submit a new issue.
```
